subsys/net/l2/ieee802154/ieee802154_mgmt.c

```diff
@@ -152,9 +152,9 @@ static int ieee802154_scan(u32_t mgmt_request, struct net_if *iface,
 		}
 
 		/* Context aware sleep */
-		k_sleep(scan->duration);
-
-		if (!ctx->scan_ctx) {
+		if (ctx->scan_ctx) {
```

Comment:
is `scan` always the same as `ctx->scan_ctx`? I notice that `scan` is itself a parameter to this function, it's a cast from the `data` parameter

Reply:
yes, it is assigned to ctx->scan_ctx in line 116.

Reply:
It looks like the `scan` parameter should never be NULL as then we would not be able to do a scan in the first place. So probably easiest solution is to check that in the beginning of the function and just return. @tbursztyka do you agree?

Reply:
Yes that's a wrong fix.

Just check scan value above (it's a cast of data pointer) after line 98. If NULL then return -EINVAL

Reply:
ok, this change this function behaviour. Should then I remove the whole:
```
if (!ctx->scan_ctx) {
        NET_DBG("Scan request cancelled");
        ret = -ECANCELED;
	break;
}
```
Since `ctx->scan_ctx` is guaranteed not NULL ?

Reply:
The idea of the scan function is that user might cancel the scan while we are waiting results in line 155 k_sleep(). So if the scan is cancelled, the ctx->scan_ctx will be set to NULL by other thread and we notice it after the sleep is finished. This means that we must not remove this block.

Reply:
I see, will keep it.